material-ui/app/entry.client.tsx

```diff
@@ -0,0 +1,38 @@
+import * as React from 'react';
+import { useState } from 'react';
+import { hydrate } from 'react-dom';
+import { RemixBrowser } from '@remix-run/react';
+import { CacheProvider } from '@emotion/react';
+import { ThemeProvider } from '@mui/material/styles';
+import CssBaseline from '@mui/material/CssBaseline';
+import ClientStyleContext from './src/ClientStyleContext';
+import createEmotionCache from './src/createEmotionCache';
+import theme from './src/theme';
+
+interface ClientCacheProviderProps {
+  children: React.ReactNode;
+}
+function ClientCacheProvider({ children }: ClientCacheProviderProps) {
```

Comment:
You can use the `PropsWithChildren` generic here

```suggestion
function ClientCacheProvider({ children }: PropsWithChildren) {
```